Change install directories to use GNUInstallDirs

Shelnutt2 opened the pull request:
This changes the install directories to use `CMAKE_INSTALL_INCLUDESDIR`, `CMAKE_INSTALL_LIBDIR` and `CMAKE_INSTALL_BINDIR` to handle linux multi-arch distros.

Closes #601

tdenniston commented:
Nice, this seems much easier than the previous approach. One question: in `FindTBB_EP.cmake` we are installing the TBB dynamic library to a bare `lib` destination as well. We probably need this fix there as well.

Shelnutt2 commented:
Good catch on the TBB install directories. I've pushed another commit to use the GNUInstallDirs for `FINDTBB_EP.cmake`. GNUInstallDirs is only valid after a project has been declared, so I've had to include it in `FINDTBB_EP.cmake` and in the `tiledb/CMakeLists.txt`. If you use it in the top level CMakeLists.txt of the project it will not set the paths correctly.

tdenniston commented:
That's fine. If you squash the two commits into a single commit, we can wait for the CI to pass and I'll merge this.

Shelnutt2 commented:
I've squashed the commit and rebased.

As a note it seems the HDFS test is failing a bit this morning due to travis not being able to reach apache.org. This test might need to be manually restarted for the pull request.

tdenniston commented:
Thanks @Shelnutt2!
